aspnet/mvc/views/razor.rst

```diff
@@ -1,8 +1,732 @@
+
+          WriteLiteral("/r/n<div>Output: ");
+          Write(output);
+          WriteLiteral("</div>");
+      }
+  }
+
+:ref:`razor-customcompilationservice-label` explains how to view this generated class. Understanding how Razor generates code for a view will make it easier to understand how directives work.
+
+``@using``
+^^^^^^^^^^^^^
+
+The ``@using`` directive will add the c# ``using`` directive to the generated razor page:
+
+.. literalinclude:: razor/sample/Views/Home/Contact9.cshtml
+  :language: html
+
+``@model``
+^^^^^^^^^^^^
+
+The ``@model`` directive allows you to specify the type of the model passed to your Razor page. It uses the following syntax:
+
+.. code-block:: none
+
+  @model TypeNameOfModel<TModel>
+
+For example, if you create a new ASP.NET Core MVC app with individual user accounts, the *Views/Account/Login.cshtml* Razor view file contains the follow model declaration:
+
+.. code-block:: c#
+
+  @model LoginViewModel
+
+In the class example in :ref:`razor-directives-label`, the class generated inherits from ``RazorPage<dynamic>``. By adding an ``@model`` you control what’s inherited. For example
+
+.. code-block:: c#
+
+  @model LoginViewModel
+
+Generates the following class
+
+.. code-block:: c#
+
+ public class _Views_Account_Login_cshtml : RazorPage<LoginViewModel>
+
+This allows you to access the strongly typed model in your Razor page through the ``Model`` property:
+
+.. code-block:: none
+
+  <div>The Login Email: @Model.Email</div>
+  
+Obviously you must pass the model from the controller to the view. See :ref:`strongly-typed-models-keyword-label` for more information.
+
+``@inherits``
+^^^^^^^^^^^^^^^
+
+The ``@inherits`` directive is similar to the model directive. ``@inherits`` gives you full control of the class your Razor page inherits:
+
+.. code-block:: none
+
+ @inherits TypeNameOfClassToInheritFrom
+
+For instance, let’s say we had the following custom Razor page type:
+
+.. literalinclude:: razor/sample/Classes/CustomRazorPage.cs
+  :language: c#
+
+The following Razor would generate ``<div>Custom text: Hello World</div>``.
+
+.. literalinclude:: razor/sample/Views/Home/Contact10.cshtml
+  :language: html
+
+The ``@inherits`` keyword is not allowed on the same page with the ``@model`` statement. You can pass a model as shown with the following Razor page (when passed "Rick@contoso.com" in the model):
+
+.. literalinclude:: razor/sample/Views/Home/Login1.cshtml
+  :language: html
+  :lines: 1-
+
+Generates this HTML:
+
+.. code-block:: none
+
+  <div>The Login Email: Rick@contoso.com</div>
+  <div>Custom text: Hello World.</div>
+
+.. review: Adding the model to _ViewImports is not needed. We don't need it to pass a model.
+
+While you can't use ``@model`` and ``@inherits`` on the same page, you can have ``@model`` in a *_ViewImports.cshtml* file that the Razor page imports. See :doc:`/mvc/views/layout`. For example, if your Razor view imported the following *_ViewImports.cshtml* file:
```

Comment:
It's a corner case only because people don't typically use inherits. However, in the case that someone does use inherits it's quite likely that they'll use this pattern.